android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java

```diff
@@ -99,10 +101,88 @@
         return accountTestData;
     }
 
+    @Test
+    public void test_compare() throws InvalidUriException, SerializationException {
+        Uri fogUri = Uri.parse("fog://some-test-uri");
+        Uri differentFogUri = Uri.parse("fog://some-other-test-uri");
+        Uri fogUriWithPort = Uri.parse("fog://some-test-uri:443");
+        TestFogConfig fogConfig = Environment.getTestFogConfig();
+        RistrettoPrivate key1 = RistrettoPrivate.generateNewKey();
+        RistrettoPrivate key2 = RistrettoPrivate.generateNewKey();
+
+        RistrettoPrivate key1_copy = RistrettoPrivate.fromBytes(key1.getKeyBytes());
+        RistrettoPrivate key2_copy = RistrettoPrivate.fromBytes(key2.getKeyBytes());
+
+        AccountKey first = new AccountKey(
+                key1,
+                key2,
+                fogUri,
+                fogConfig.getFogReportId(),
+                fogConfig.getFogAuthoritySpki()
+        );
+        try {
```

Comment:
What's the advantage of writing it this way as opposed to "Assert.assertNotNull(first)"?

Reply:
`first` is valid already since it's returned from the NonNull API, but I'm testing `equals` method to be able to compare to `null` without any exceptions